Hamilton_Matthew/detail.html

```diff
@@ -0,0 +1,138 @@
+<%inherit file="app_base.htm" />
+<%! from catalog import models as cmod %>
+
+<%block name="content_top">
+    <h1 class = "myh1">${ pagetitle }</h1>
+</%block>
+
+        <%block name="content_middle">
+            <ul>
+            <div class ="col-md-1">
+                    % for i in images:
+                        <li>  
+                            <img src = ${i} class = "left_pane_image" alt = ${i}>
+                        </li>
+                    % endfor
+                </div>
+            </ul>
+            <div class="col-md-3">
+                
+                    <img src="${product.image_url()}" class = "prod_img" alt = "${product.name}" >
+                    <!-- <img src="${product.image_url()}" onmouseover = "hover(this);" onmouseout="unhover(this);" /> -->
+                    <!-- <img src = "https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcRd7uQ4hV5W8S1bMQmiY5CIA3nqL1ExWyrhOuZJZzNy-wUDEjCqxw" class = "prod_img"> -->
+
+            </div>
+            
+            <div class="col-md-4">
+                
+                <p>${product.name | h}</p>
+                <br>
+                <p>${product.category | h}</p>
+                <br>
+                <p>${product.__class__.__name__ | h}</p>
+                <br>
+                <p>$${product.price | h}</p>
+                <br>
+                <p>${product.description}</p>
+                <br>
+                <br>
+                <br>
+                % if product.__class__.__name__ == "BulkProduct":
+                    <input type="number" class="form-control quantity_input" placeholder = ${product.quantity | h}> &nbsp &nbsp
+                % endif
+                <button class = "btn btn-buy">Buy</button>
+                <br>
+                <br>
+            </div>      
+        </%block>
+
+<%block name = "content_right">   
+    <div class="col-md-2">
+      <h2 class = "myh2">Recently Viewed</h2>
+      <br/>
+
+      <ol class = "right_list">
+
+    % if int(request.session['new_prod']) in request.session['last_five'][1:6]:
+        Sup fam
+        <%
+            print("finding index, then prod_id")
+            index = request.session['last_five'].index(request.session['new_prod'])
+            print(index)
+
+            my_prod = request.session['last_five'][index]
+            print(my_prod)
+
+            print(request.session['last_five'])
+            request.session['last_five'].remove(my_prod)
+            print(request.session['last_five'])
+
+            request.session['last_five'].insert(0, my_prod)
+            print(request.session['last_five'])
+
+            request.last_five = []
+
+            for product_id in request.session['last_five']:
+                request.last_five.append(cmod.Product.objects.get(id = product_id))
+        %>
+    % endif
```

Comment:
This block would better in detail.py so it is taken care of on the server rather than the browser. Also, remember to remove the printing before merging to master.

Reply:
I moved it to detail.py and will do